prover/prover_cli/src/commands/status/batch.rs

```diff
@@ -50,128 +63,351 @@ async fn get_batches_data(
+                _ => (),
+            }
+        }
+    }
+}
+
+fn display_leaf_witness_generator_jobs_info(
+    mut leaf_witness_generators_jobs_info: Vec<LeafWitnessGeneratorJobInfo>,
+) {
+    leaf_witness_generators_jobs_info.sort_by_key(|job| job.circuit_id);
+
+    leaf_witness_generators_jobs_info.iter().for_each(|job| {
+        println!(
+            "   > {}: {}",
+            format!(
+                "{:?}",
+                BaseLayerCircuitType::from_numeric_value(job.circuit_id as u8)
+            )
+            .bold(),
+            Status::from(job.status.clone())
+        )
+    });
+}
+
+fn display_node_witness_generator_jobs_info(
+    mut node_witness_generators_jobs_info: Vec<NodeWitnessGeneratorJobInfo>,
+) {
+    node_witness_generators_jobs_info.sort_by_key(|job| job.circuit_id);
+
+    node_witness_generators_jobs_info.iter().for_each(|job| {
+        println!(
+            "   > {}: {}",
+            format!(
+                "{:?}",
+                BaseLayerCircuitType::from_numeric_value(job.circuit_id as u8)
+            )
+            .bold(),
+            Status::from(job.status.clone())
+        )
+    });
+}
+
+fn display_prover_jobs_info(prover_jobs_info: Vec<ProverJobFriInfo>) {
+    let prover_jobs_status = Status::from(prover_jobs_info.clone());
+
+    if matches!(prover_jobs_status, Status::Successful) {
+        println!(
+            "> {}: {prover_jobs_status}",
+            "Prover Jobs".to_owned().bold()
+        );
+        return;
+    }
+
+    println!(
+        "v {}: {prover_jobs_status}",
+        "Prover Jobs".to_owned().bold()
+    );
+
+    let mut jobs_by_circuit_id: HashMap<u32, Vec<ProverJobFriInfo>> = HashMap::new();
+
+    prover_jobs_info.iter().for_each(|job| {
+        jobs_by_circuit_id
+            .entry(job.circuit_id)
+            .or_default()
+            .push(job.clone())
+    });
+
+    let mut jobs_by_circuit_id: Vec<(u32, Vec<ProverJobFriInfo>)> = jobs_by_circuit_id
+        .iter()
+        .map(|(key, value)| (*key, value.clone()))
+        .collect();
+
+    jobs_by_circuit_id.sort_by_key(|job| job.0);
+
+    for (circuit_id, prover_jobs_info) in jobs_by_circuit_id {
+        let status = Status::from(prover_jobs_info.clone());
+        match status {
+            Status::InProgress => {
+                println!(
+                    "   > {}: {}",
+                    format!(
+                        "{:?}",
+                        BaseLayerCircuitType::from_numeric_value(circuit_id as u8)
+                    )
+                    .bold(),
+                    status
+                );
+                display_job_status_count(prover_jobs_info);
+            }
+            _ => println!(
+                "   > {}: {}",
+                format!(
+                    "{:?}",
+                    BaseLayerCircuitType::from_numeric_value(circuit_id as u8)
+                )
+                .bold(),
+                status
+            ),
+        };
```

Comment:
I believe the only difference between cases here is the `display_job_status_count` call. This is a more brief way to express the same:
```suggestion
        println!(
            "   > {}: {}",
            format!("{:?}", BaseLayerCircuitType::from_numeric_value(circuit_id as u8)).bold(),
            status
        );
        if matches!(status, Status::InProgress) {
                display_job_status_count(prover_jobs_info);
        }
```